### PR TITLE
Simplify annotation in `_cmaes.py`

### PR DESCRIPTION
**Pull request by RektPunk**
## Motivation
Related to https://github.com/optuna/optuna/issues/4508

## Description of the changes
I removed the calls to the typing module except for `Any`, `cast` and `TYPE_CHECKING`.

### CODE REVIEW

**Review by eukaryo — APPROVED** [review]
LGTM!

**Comment by eukaryo** [comment]
@porink0424 , could you review this PR?

File: optuna/samplers/_cmaes.py [hunk 1]
```diff
@@ -35,8 +30,6 @@
 
 if TYPE_CHECKING:
     import cmaes
-
-    CmaClass = Union[cmaes.CMA, cmaes.SepCMA, cmaes.CMAwM]
```
[the remark this answers is not in the document]
Comment (c-bata) [reply optuna/samplers/_cmaes.py:39]: Thank you for asking. I agree with @eukaryo.  @RektPunk Could you keep `CmaClass` as it is?

[the remark this answers is not in the document]
Comment (RektPunk) [reply optuna/samplers/_cmaes.py]: Sure, it is reverted in fd71096. Thanks for reviewing, btw.

**Review by porink0424 — APPROVED** [review]
LGTM!

**Pull request merged**